### PR TITLE
Reader: use subscription ID rather than feed URL to unsubscribe wherever possible

### LINKED ISSUE #945: Reader: cannot unfollow RSS feed
Raised by @alternatekev 

A long time ago, I subscribed to a Tumblr blog using our Reader. The blog is http://frakintosh.com

I have never been able to unsubscribe from it, as I'd prefer to just use Tumblr for Tumblr blogs since we can't interact with them on WordPress.com. Here's what happens when I try to unsubscribe:

1. Load up the Reader
2. See new post from this blog's RSS feed
3. Click on the ellipsis menu
4. Click unsubscribe
5. Reload the page, click ellipsis menu...
6. Still subscribed.

Video: https://cloudup.com/cmSnuS5wnuJ

This also doesn't work when viewing the blog's Reader page.

I found some closed issues/PRs that appear to fix this issue by using the blog ID vs the blog's URL, but it hasn't fixed the unfollow bug I describe above.

Another report raised by @aerych 

To reproduce:

1. Edit your following sites and follow "ma.tt", "gizmodo.com", and "arstechnica.com"
View posts for "Followed Sites"

2. Find a post from ma.tt open the menu and choose unfollow.  Refresh the list (or return to the manage subscriptions page) and confirm that you are no longer following.  Everything works as it should.

3. Return to the post list and unfollow a post from either gizmodo.com or arstechnica.com. 
The menu updates as if you were no longer following.
Refresh the list (or return to the manage subscriptions page) and confirm that you are no longer following.  The unfollow did not work. 

It looks like external feeds without Jetpack might be the only ones affected.

Notes from @blowery 

> My guess is that we're unfollowing by the blog URL, not the feed URL, and when we try to unfollow we don't find the site. We should really switch to unfollowing by feed id, not URL, as that takes all the guesswork out.
> 
> This requires modifying the API to take either a feed ID or a URL.

Related: #749

### PR DESCRIPTION
**Pull request by bluefuton**
The capability to unsubscribe from feeds using subscription ID (sub_id) rather than feed URL was recently added to the `/read/following/mine/delete` endpoint in r129630-wpcom

This PR switches all feed unsubscriptions to use the more reliable subscription ID wherever possible.

Fixes #945 and #2064.

### CODE REVIEW

**Comment by bluefuton** [comment]
I'll add this to Manage Following initially.

The tricky part is dealing with the `FeedSubscriptionStore`, which runs entirely off feed URL at the moment.

**Comment by blowery** [comment]
We'll have to update all the follow buttons too, to take either the feed ID or the blog ID.

**Comment by blowery** [comment]
We probably need a ways to unsub by sub ID, feed ID, and blog ID. The latter two would look up the sub and then unsub by sub ID.

File: client/lib/reader-feed-subscriptions/index.js [hunk 1]
```diff
@@ -1,381 +1,170 @@
...
+	const state = FeedSubscriptionStore.get();
+	return state.get( 'isFetching' );
+};
+
+FeedSubscriptionStore.getLastError = function() {
+	const state = FeedSubscriptionStore.get();
+	return state.has( 'errors' ) ? state.get( 'errors' ).last() : null;
+};
+
+FeedSubscriptionStore.isLastPage = function() {
```
Comment (blowery) [inline client/lib/reader-feed-subscriptions/index.js]: if we go es6, these could be `() => FeedSubscriptionStore.get().get( 'isLastPage' )`

File: client/lib/reader-feed-subscriptions/index.js [hunk 2]
```diff
@@ -1,381 +1,170 @@
...
+
+		case actionTypes.UNFOLLOW_READER_FEED:
+			return removeSubscription( state, payload.action.data );
+	}
+
+	return state;
+}, initialState );
+
+FeedSubscriptionStore.isFetching = function() {
+	const state = FeedSubscriptionStore.get();
```
Comment (blowery) [inline client/lib/reader-feed-subscriptions/index.js]: is there any reason to expose the whole state and have getter? maybe we just hide it in a module var and use it direct. Then we could can FeedSubscriptionStore, the object, and just export the useful functions.

**Comment by bluefuton** [comment]
Todo: 
- [ ] handle `DISMISS_FOLLOW_ERROR`
- [ ] handle `RECEIVE_FEED_POST`
- [ ] handle `RECEIVE_FEED_SUBSCRIPTIONS`
- [ ] find an alternative way of handling new subscription event (currently emit an 'add' event from store)
- [ ] alter follow button to accept a sub ID
- [ ] alter Manage Following to use sub ID instead of URL for unfollow

File: client/lib/reader-feed-subscriptions/index.js (hunk 1 above)
Comment (bluefuton) [inline client/lib/reader-feed-subscriptions/index.js]: :+1:

**Comment by bluefuton** [comment]
I think this PR has been a useful exercise, but #3059 addresses our immediate concerns with unfollows. I think perhaps we're better off porting the store functionality straight to Redux.

**Pull request closed without merging**